### PR TITLE
fix: set-cookie header for tencent api gateway

### PR DESCRIPTION
**Pull request by yugasun**
Relative issue https://github.com/serverless-components/tencent-express/issues/6

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 npm-debug.log
 /package-lock.json
 tmp/
+.DS_Store
```
Comment (medikoo) [inline .gitignore]: This should rather be configured in user global ignore (this file is intentionally pretty simple, and contains only ignores that are directly related to project.

See: https://eshlox.net/2017/08/04/git-global-gitignore-file

**Review by medikoo — COMMENTED** [review]
@yugasun instead of copy & paste. It'll be good to put that logic in one place (some external package?) and reuse it in both koa and express components.

Otherwise we're likely to run into maintenance issues

**Review by pmuens — COMMENTED** [review]
Thanks for working on this @yugasun :+1:

I agree with @medikoo here. DRYing this out will make things easier in the future.

**Comment by yugasun** [comment]
> @yugasun instead of copy & paste. It'll be good to put that logic in one place (some external package?) and reuse it in both koa and express components.
> 
> Otherwise we're likely to run into maintenance issues

Ok, I will create a fork for `aws-serverless-express` module.

**Comment by yugasun** [comment]
@pmuens @medikoo  I use [tencent-serverless-nodejs](https://github.com/yugasun/tencent-serverless-nodejs) for http proxy. Could you help to review again?

File: package.json [hunk 2]
```diff
@@ -15,13 +15,14 @@
...
-    "@serverless/tencent-scf": "^2.0.2",
-    "aws-serverless-express": "^3.3.6",
+    "@serverless/tencent-scf": "^2.0.5",
     "ext": "^1.4.0",
     "fs-extra": "^8.1.0",
     "prettier": "^1.19.1",
+    "tencent-serverless-nodejs": "^1.0.0",
     "terser": "^4.4.2",
     "type": "^2.0.0",
+    "type-is": "^1.6.18",
```
Comment (medikoo) [inline package.json]: I think this dependency is no longer needed (?)

**Review by medikoo — COMMENTED** [review]
@yugasun looks good. Just wondering why external package was named `tencent-serverless-nodejs`, name seems too abstract (doesn't give a hint it's about proxying HTTP which serves projects as `express`, `connect` or `koa`), but it's not a big issue ofc :)

**Comment by yugasun** [comment]
@medikoo I call it `tencent-serverless-nodejs`, because It can be used for all nodejs framework, like express.js, koa.js, egg.js.....

File: package.json (hunk 2 above)
Comment (yugasun) [reply package.json]: Yes, I had removed it.

**Review by medikoo — DISMISSED** [review]
> @medikoo I call it tencent-serverless-nodejs, because It can be used for all nodejs framework, like express.js, koa.js, egg.js.....

Sure, still Node.js on its own is way more than those frameworks. There are complex Node.js programs which do not involve HTTP servers at all.

I guess more accurate name would be `tencent-serverless-http`

**Review by pmuens — DISMISSED** [review]

**Review by pmuens — APPROVED** [review]

**Pull request merged**